CUDADataFormats/HGCal/interface/HGCConditions.h

```diff
@@ -0,0 +1,197 @@
+  }
+  constexpr int32_t waferX() { return (-2 * waferU() + waferV()); }
+  constexpr int32_t waferY() { return (2 * waferV()); }
+  constexpr int32_t cellU() { return (id_ >> kHGCalCellUOffset) & kHGCalCellUMask; }
+  constexpr int32_t cellV() { return (id_ >> kHGCalCellVOffset) & kHGCalCellVMask; }
+  constexpr int32_t nCellsSide() { return (type() == HGCalFine) ? HGCalFineN : HGCalCoarseN; }
+  constexpr int32_t cellX() {
+    const int32_t N = nCellsSide();
+    return (3 * (cellV() - N) + 2);
+  }
+  constexpr int32_t cellY() {
+    const int32_t N = nCellsSide();
+    return (2 * cellU() - (N + cellV()));
+  }
+
+private:
+  uint32_t id_;
+  enum waferType { HGCalFine = 0, HGCalCoarseThin = 1, HGCalCoarseThick = 2 };
+  static const int32_t HGCalFineN = 12;
+  static const int32_t HGCalCoarseN = 8;
+  static const int32_t kHGCalCellUOffset = 0;
+  static const int32_t kHGCalCellUMask = 0x1F;
+  static const int32_t kHGCalCellVOffset = 5;
+  static const int32_t kHGCalCellVMask = 0x1F;
+  static const int32_t kHGCalWaferUOffset = 10;
+  static const int32_t kHGCalWaferUMask = 0xF;
+  static const int32_t kHGCalWaferUSignOffset = 14;
+  static const int32_t kHGCalWaferUSignMask = 0x1;
+  static const int32_t kHGCalWaferVOffset = 15;
+  static const int32_t kHGCalWaferVMask = 0xF;
+  static const int32_t kHGCalWaferVSignOffset = 19;
+  static const int32_t kHGCalWaferVSignMask = 0x1;
+  static const int32_t kHGCalLayerOffset = 20;
+  static const int32_t kHGCalLayerMask = 0x1F;
+  static const int32_t kHGCalZsideOffset = 25;
+  static const int32_t kHGCalZsideMask = 0x1;
+  static const int32_t kHGCalTypeOffset = 26;
+  static const int32_t kHGCalTypeMask = 0x3;
+};
+
+class HeterogeneousHGCScintillatorDetId {
+public:
+  constexpr HeterogeneousHGCScintillatorDetId(uint32_t id) : id_(id) {}
+  constexpr int type() { return (id_ >> kHGCalTypeOffset) & kHGCalTypeMask; }
+  constexpr int zside() const { return (((id_ >> kHGCalZsideOffset) & kHGCalZsideMask) ? -1 : 1); }
+  constexpr int layer() const { return (id_ >> kHGCalLayerOffset) & kHGCalLayerMask; }
+
+private:
+  uint32_t id_;
+  uint32_t kHGCalPhiOffset = 0;
+  uint32_t kHGCalPhiMask = 0x1FF;
+  uint32_t kHGCalRadiusOffset = 9;
+  uint32_t kHGCalRadiusMask = 0xFF;
+  uint32_t kHGCalLayerOffset = 17;
+  uint32_t kHGCalLayerMask = 0x1F;
+  uint32_t kHGCalTriggerOffset = 22;
+  uint32_t kHGCalTriggerMask = 0x1;
+  uint32_t kHGCalZsideOffset = 25;
+  uint32_t kHGCalZsideMask = 0x1;
+  uint32_t kHGCalTypeOffset = 26;
+  uint32_t kHGCalTypeMask = 0x3;
+};
+
+namespace hgcal_conditions {
+  namespace parameters {
+    enum class HeterogeneousHGCalEEParametersType { Double, Int32_t };
+    enum class HeterogeneousHGCalHEFParametersType { Double, Int32_t };
+    enum class HeterogeneousHGCalHEBParametersType { Double, Int32_t };
+
+    const std::vector<HeterogeneousHGCalEEParametersType> typesEE = {HeterogeneousHGCalEEParametersType::Double,
```

Comment:
if `typesEE` size is fixed, consider using an `std::array` instead of an `std::vector`